pymc/sampling/mcmc.py

```diff
@@ -168,6 +168,11 @@ def assign_step_methods(model, step=None, methods=None, step_kwargs=None):
         except TypeError:
             steps.append(step)
         for step in steps:
+            for var in step.vars:
+                if var not in model.value_vars:
+                    raise ValueError(
+                        "Step samplers should only contain value variables that belong to the model."
```

Comment:
Maybe something more informative?

```suggestion
                        f"{var} assigned to {step} sampler is not a value variable in the model. 
                        You can use `util.get_value_vars_from_user_vars` to parse user provided variables.
```

Reply:
the phrasing was changed as suggested